### ISSUE TITLE
Add more control of animation with code api for make anim without animgraph

### ISSUE THREAD

**Issue by dictateurfou** [Facepunch/sbox-issues#4075]
### For?

S&Box

### What can't you do?

I wanted to create a certain furniture system with drawers or doors that can be opened, I would have liked to do it without the animgraph because there is no need for a complex tool to do simple things.

Currently the problem is that it is impossible to play multiple animations at the same time with code. We can only play one with "CurrentSequence" 

so if a piece of furniture has 2 doors this poses a problem and I am therefore obliged to use an animgraph if I want to control each door independently

### How would you like it to work?

maybe add a way to access the AnimationSequence class or be able to add a new sequence in a list of sequence

### What have you tried?

I tried something but going back into the classes I noticed that the CurrentSequence variable was not reproducible because of the internal functions

**Comment by MrBrax** [comment]
SetBoneTransform?

**Comment by dictateurfou** [comment]
> SetBoneTransform?
that could be a solution. But that's not really one. I would have to manage the annimations with code I would therefore lose the interest of the annimations, I would have to make a transition at each tick to make a real annimation and not a tp and suppose that I need to make more complex animations I don't even know if I would have access to the data of the anim without playing it but I would only reintegrate what the game already does which would have no interest as I might as well use the game tools directly xD

**Comment by MrBrax** [comment]
You can use lerps with rotations and vectors to make a nice sliding animation for drawers/doors but i get what you mean.

**Comment by dictateurfou** [comment]
yes I know but I am making a generic class for so I would rather use the annimations which are in my model, because otherwise I will have to configure everything ig, store them in parameters for each model, and just have a cabinet which opens in 2 parts forming a triangle with the door it will be difficult to configure whereas on blender I would just have to move a bone xD.

Afterwards I agree with you it is possible to tinker with something to replace the animations and make a module to transform the bones directly ig to make the animations the only drawback would be that I would not be able to store the data of the anims on the vmodel could be a good experience but I would prefer to advance my gameplay than do it xD
